gym_solo/core/test_termination_factory.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+from gym_solo.core import termination
+
+# TODO: Move this to gym_solo.testing
+class DummyTermination(termination.Termination):
+  def __init__(self, body_id: int, termination_var: bool):
+    self.body_id = body_id
+    self.termination_var = termination_var
+    self.reset_counter = 0
+    self.reset()
+    
+  def reset(self):
+    self.reset_counter += 1
+
+  def is_terminated(self) -> bool:
+    return self.termination_var
+
+class TestTerminationFactory(unittest.TestCase):
+  def test_initialization(self):
+    termination_factory = termination.TerminationFactory()
+    self.assertFalse(termination_factory._terminations)
+    self.assertTrue(termination_factory._use_or)
+
+  def test_register_termination(self):
+    termination_factory = termination.TerminationFactory()
+    dummy_termination_1 = DummyTermination(0, True)
+    dummy_termination_2 = DummyTermination(0, False)
+    termination_factory.register_termination(dummy_termination_1, 
+      dummy_termination_2)
```

Comment:
If i really wanted to nit, you don't need to make the `dummy_termination_*` variables (just instantiate them in the method call). however, I just had to make another comment cause I'm literally nutting my pants cause of this argument expansion!! 😝

Reply:
I might be just missing all the context here @agupta231. Which argument expansion are you talking about?

Reply:
lmfao i'm stoopid :)